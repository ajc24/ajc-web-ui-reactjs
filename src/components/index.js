/**
 * Developed by Anthony Cox in 2024
 */

/* Export the header components */
export { default as HeaderTypeA } from './header/HeaderTypeA';

/* Export the main components */
export { default as MainContent } from './main/MainContent';

/* Export the menu bar components */
export { default as MenuBarTypeA } from './menu-bar/MenuBarTypeA';
