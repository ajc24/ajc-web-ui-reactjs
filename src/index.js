/**
 * Developed by Anthony Cox in 2024
 */

/* Import the header components */
import Header from './components/header/Header';

/* Import the image components */
import DecorativeImage from './components/images/DecorativeImage';
import FunctionalImage from './components/images/FunctionalImage';
import InformativeImage from './components/images/InformativeImage';

/* Export all components */
export {
  DecorativeImage,
  FunctionalImage,
  InformativeImage,
  Header,
};
