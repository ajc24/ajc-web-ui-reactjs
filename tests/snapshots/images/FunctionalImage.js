/**
 * Developed by Anthony Cox in 2024
 */
import 'jsdom-global/register';
import React from 'react';
import { TestDev } from 'ajc-testing-code';
import { FunctionalImage } from '../../../src';
import storybookLogo from '../../../stories/images/files/storybook-logo.svg';

describe('Functional Image', () => {
  describe('With default parameters', () => {
    let snapshot;
    
    beforeAll(() => {
      snapshot = TestDev.createSnapshot(
        <React.Fragment>
          <FunctionalImage id="functional-image-1" src={storybookLogo} alt="Default functional component alt text" onClick={() => {}} />
        </React.Fragment>
      );
    });

    it('verifies the snapshot for the component', () => {
      expect(snapshot).toMatchSnapshot();
    });
  });

  describe('With custom parameters', () => {
    let snapshot;
    
    beforeAll(() => {
      snapshot = TestDev.createSnapshot(
        <React.Fragment>
          <FunctionalImage id="functional-image-2" src={storybookLogo} alt="Custom functional component alt text" onClick={() => {}} height={300} width={500} />
        </React.Fragment>
      );
    });

    it('verifies the snapshot for the component', () => {
      expect(snapshot).toMatchSnapshot();
    });
  });
});
