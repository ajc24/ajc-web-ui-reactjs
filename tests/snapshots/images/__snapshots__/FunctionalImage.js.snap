// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Functional Image With custom parameters verifies the snapshot for the component 1`] = `"<img id="functional-image-2--functional-image" class="image image-animations" width="500" height="300" alt="Custom functional component alt text" src="[object Object]">"`;

exports[`Functional Image With default parameters verifies the snapshot for the component 1`] = `"<img id="functional-image-1--functional-image" class="image image-animations" width="100" height="100" alt="Default functional component alt text" src="[object Object]">"`;
